### DISCUSSION TITLE
First command may fail, the second command should run

### DISCUSSION THREAD

**Discussion by manuelesimi** [nextflow-io/nextflow#2536] [Q&A] [answered]
Could I please have some help with a workaround from someone who is more experienced with bash and Nextflow? I have a process that has two commands, one that can fail and another that should run even if the first fails. The above trap implementation exits before the second command so I need something that can suppress or change the exit code like so
```nextflow
process foo {
  '''
  trap 'if [[ $? == expected_error ]]; then echo "OK supreesed error"; fi' EXIT
  your_command_that_may_fail
  
  second_command
  '''
} 
```

_Originally posted by @NickSwainston in https://github.com/nextflow-io/nextflow/issues/1564#issuecomment-1005367117_

**Comment by manuelesimi** [comment, marked as answer]
@NickSwainston

This should do it: 
```
process foo {
  '''
  your_command_that_may_fail || true
  
  second_command
  '''
} 
```

**Reply by NickSwainston** [reply]
Yup, that works! Thank you for your quick response.
